tests/test_basic_relay.go

```diff
@@ -0,0 +1,283 @@
+
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/utils/logging"
+	avalancheWarp "github.com/ava-labs/avalanchego/vms/platformvm/warp"
+	"github.com/ava-labs/awm-relayer/config"
+	"github.com/ava-labs/awm-relayer/database"
+	"github.com/ava-labs/awm-relayer/messages/teleporter"
+	"github.com/ava-labs/awm-relayer/peers"
+	testUtils "github.com/ava-labs/awm-relayer/tests/utils"
+	"github.com/ava-labs/subnet-evm/core/types"
+	predicateutils "github.com/ava-labs/subnet-evm/utils/predicate"
+	warpPayload "github.com/ava-labs/subnet-evm/warp/payload"
+	"github.com/ava-labs/subnet-evm/x/warp"
+	teleporterTestUtils "github.com/ava-labs/teleporter/tests/utils"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+var (
+	relayerConfigPath string
+	teleporterMessage = teleporter.TeleporterMessage{
+		MessageID:               big.NewInt(1),
+		SenderAddress:           teleporterTestUtils.FundedAddress,
+		DestinationAddress:      teleporterTestUtils.FundedAddress,
+		RequiredGasLimit:        big.NewInt(1),
+		AllowedRelayerAddresses: []common.Address{},
+		Receipts:                []teleporter.TeleporterMessageReceipt{},
+		Message:                 []byte{1, 2, 3, 4},
+	}
+	storageLocation = fmt.Sprintf("%s/.awm-relayer-storage", os.TempDir())
+)
+
+// Ginkgo describe node that acts as a container for the relayer e2e tests. This test suite
+// will run in order, starting off by setting up the subnet URIs and creating a relayer config
+// file. It will then build the relayer binary and run it with the config file. The relayer
+// will then send a transaction to the source subnet to issue a Warp message simulting a transaction
+// sent from the Teleporter contract. The relayer will then wait for the transaction to be confirmed
+// on the destination subnet and verify that the Warp message was received and unpacked correctly.
+var _ = ginkgo.Describe("[Relayer E2E]", ginkgo.Ordered, func() {
+	var (
+		receivedWarpMessage *avalancheWarp.Message
+		payload             []byte
+		relayerCmd          *exec.Cmd
+		relayerCancel       context.CancelFunc
+	)
+
+	ginkgo.It("Set up relayer config", ginkgo.Label("Relayer", "Setup Relayer"), func() {
+		hostA, portA, err := teleporterTestUtils.GetURIHostAndPort(teleporterTestUtils.ChainANodeURIs[0])
+		Expect(err).Should(BeNil())
+
+		hostB, portB, err := teleporterTestUtils.GetURIHostAndPort(teleporterTestUtils.ChainBNodeURIs[0])
+		Expect(err).Should(BeNil())
+
+		log.Info(
+			"Setting up relayer config",
+			"hostA", hostA,
+			"portA", portA,
+			"blockChainA", teleporterTestUtils.BlockchainIDA.String(),
+			"hostB", hostB,
+			"portB", portB,
+			"blockChainB", teleporterTestUtils.BlockchainIDB.String(),
+			"teleporterTestUtils.SubnetA", teleporterTestUtils.SubnetA.String(),
+			"teleporterTestUtils.SubnetB", teleporterTestUtils.SubnetB.String(),
+		)
+
+		relayerConfig := config.Config{
```

Comment:
Some of the fields (for example `APINodeHost` and `APINodePort` are generated dynamically by the test, so we'd have to modify this variable in the test anyway. Since it's only used in one place, I think we should leave it as is. As the relayer test suite expands, we can consider ways to reduce redundant setup code.